packages/@aws-cdk/aws-iam/README.md

```diff
@@ -364,6 +364,47 @@ const provider = new iam.OpenIdConnectProvider(this, 'MyProvider', {
 const principal = new iam.OpenIdConnectPrincipal(provider);
 ```
 
+## SAML provider
+
+An IAM SAML 2.0 identity provider is an entity in IAM that describes an external
+identity provider (IdP) service that supports the SAML 2.0 (Security Assertion
+Markup Language 2.0) standard. You use an IAM identity provider when you want
+to establish trust between a SAML-compatible IdP such as Shibboleth or Active
+Directory Federation Services and AWS, so that users in your organization can
+access AWS resources. IAM SAML identity providers are used as principals in an
+IAM trust policy.
+
+```ts
+new iam.SamlProvider(this, 'Provider', {
+  metadataDocument: fs.readFileSync('/path/to/saml-metadata-document.xml', 'utf-8'),
+});
+```
+
+The `SamlPrincipal` class can be used as a principal with a `SamlProvider`:
+
+```ts
+const provider = new iam.SamlProvider(this, 'Provider', {
+  metadataDocument: fs.readFileSync('/path/to/saml-metadata-document.xml', 'utf-8'),
```

Comment:
Can we abstract this away? I don't think anybody would want to inline the XML and read from a file would be the default. So what about providing only a file name to the L2?

Reply:
In most cases you are right but it could be returned by a custom resource?

Reply:
So what about a SamlMetadata class that has fromInline and fromFile?